service/k8sclient/k8sclient.go

```diff
@@ -1259,3 +1273,130 @@ func (c *K8sClient) GetEvents(ctx context.Context, pod string) ([]string, error)
+			}
+		}
+	}
+	return workers, nil
+}
+
+// GetAllClusterResources goes through all cluster nodes and sums their allocatable resources.
+func (c *K8sClient) GetAllClusterResources(ctx context.Context) (
+	cpuMillis int64, memoryBytes int64, diskSizeBytes int64, err error,
+) {
+	nodes, err := c.getWorkerNodes(ctx)
+	if err != nil {
+		return 0, 0, 0, errors.Wrap(err, "could not get a list of nodes")
+	}
+	for _, node := range nodes {
+		cpu, ok := node.Status.Allocatable[common.ResourceCPU]
+		if !ok {
+			return 0, 0, 0, errors.Errorf(
+				"node's allocatable object does not have %s field: could not get all resources",
+				common.ResourceCPU,
+			)
+		}
+		millis, err := convertors.StrToMilliCPU(cpu)
+		if err != nil {
+			return 0, 0, 0, errors.Wrap(err, "could not get allocatable CPU of the node")
+		}
+		cpuMillis += millis
+
+		memory, ok := node.Status.Allocatable[common.ResourceMemory]
+		if !ok {
+			return 0, 0, 0, errors.Errorf(
+				"node's allocatable field does not have %s field: could not get all resources",
+				common.ResourceMemory,
+			)
+		}
+		bytes, err := convertors.StrToBytes(memory)
+		if err != nil {
+			return 0, 0, 0, errors.Wrap(err, "could not get allocatable memory of the node")
+		}
+		memoryBytes += bytes
+	}
+	return cpuMillis, memoryBytes, diskSizeBytes, nil
+}
+
+type convertor func(string) (int64, error)
+
+func getRequestsAsInt64(container common.ContainerSpec, resource common.ResourceName, convertFunc convertor) (int64, error) {
+	amount, ok := container.Resources.Requests[resource]
+	if ok {
+		value, err := convertFunc(amount)
+		if err != nil {
+			return 0, errors.Wrapf(err, "failed to convert container's %s to integer", resource)
+		}
+		return value, nil
+	}
+	return 0, nil
+}
+
+// GetConsumedResources returns consumed resources in given namespace. If namespace
+// is empty string, it tries to get consumed resouces from all namespaces.
+func (c *K8sClient) GetConsumedResources(ctx context.Context, namespace string) (
+	cpuMillis int64, memoryBytes int64, diskSizeBytes int64, err error,
+) {
+	if namespace == "" {
+		namespace = "--all-namespaces"
+	} else {
+		namespace = "-n" + namespace
+	}
+	pods, err := c.GetPods(ctx, namespace)
+	if err != nil {
+		return 0, 0, 0, errors.Wrap(err, "could not get consumed resources")
+	}
+	for _, ppod := range pods.Items {
+		if ppod.Status.Phase == common.PodPhasePending {
+			continue
+		}
+		nonTerminatedInitContainers := make([]common.ContainerSpec, 0, len(ppod.Spec.InitContainers))
+		for _, container := range ppod.Spec.InitContainers {
+			if !common.IsContainerInState(
+				ppod.Status.InitContainerStatuses, common.ContainerStateTerminated, container.Name,
+			) {
+				nonTerminatedInitContainers = append(nonTerminatedInitContainers, container)
+			}
+		}
+		for _, container := range append(ppod.Spec.Containers, nonTerminatedInitContainers...) {
+			millis, err := getRequestsAsInt64(container, common.ResourceCPU, convertors.StrToMilliCPU)
```

Comment:
sorry for misunderstanding I meant moving almost whole body of for to a separate method like
```
cpu, memory, err := getResources(container.Resources.Requests)
```
and reusing it in https://github.com/percona-platform/dbaas-controller/pull/151/files#diff-68d507467191c86afbfbd5f639a2f013e2a5a15a8050bf2af591f1cbcce1a19eR1318

Reply:
I see, I will change it.

Reply:
Implemented